pcli/src/sync.rs

```diff
@@ -36,3 +36,21 @@ pub async fn sync(state: &mut ClientStateFile, wallet_uri: String) -> Result<()>
     tracing::info!(end_height = ?state.last_block_height().unwrap(), "finished sync");
     Ok(())
 }
+
+/// Fetches the global chain parameters and stores them on `ClientState`.
+#[instrument(skip(state), fields(start_height = state.last_block_height()))]
+pub async fn set_chain_params(state: &mut ClientStateFile, wallet_uri: String) -> Result<()> {
+    tracing::info!("starting set_chain_params");
+    let mut client = LightWalletClient::connect(wallet_uri).await?;
+
+    let params = client
+        .chain_params(tonic::Request::new(ChainParamsRequest {}))
+        .await?
+        .into_inner();
+
+    let cp_mut = state.chain_params_mut();
+    *cp_mut = Some(params.into());
+    state.commit()?;
+    tracing::info!(end_height = ?state.last_block_height().unwrap(), "finished sync");
+    Ok(())
+}
```

Comment:
I think this would fit better in the `fetch` module as `fetch::chain_params`, and rather than editing the state itself, it could just fetch the params, so the caller can do whatever with them (either set them, or something else).

Reply:
Did this while resolving merge conflicts with #361